src/v/resource_mgmt/cpu_profiler.h

```diff
@@ -76,11 +76,18 @@ class cpu_profiler : public ss::peering_sharded_service<cpu_profiler> {
     // is called on.
     shard_samples shard_results() const;
 
+    ss::future<std::vector<shard_samples>> override_and_get_results(
+      std::chrono::milliseconds, std::optional<ss::shard_id> shard_id);
+
 private:
     // Used to poll seastar at set intervals to capture all samples
     ss::timer<ss::lowres_clock> _query_timer;
     ss::gate _gate;
 
+    // Used by the API to override cluster config options for a set period of
```

Comment:
nit: This is the high-level purpose, but I think this comment could be clearer on what the value represents. It's the number of currently active override requests, right?

Reply:
Good catch, this comment was actually for an older mechanism I was using to implement this. Will change it to be more specific to what is happening now.